main.py

```diff
@@ -0,0 +1,123 @@
+if open("./.botmem/warnings.json", "r").read() == "":
+    warnings = {}
+else:
+    warnings = json.loads(open("./.botmem/warnings.json").read())
+##############################
+
+@bot.event # When the bot first loads up
+async def on_ready():
+    print(""" _______
+||  _  ||
+|| | | || version {} """.format(botver))
+    print('The bot has logged in as {0.user}'.format(bot)) # Say that [bot name] is logged on in the terminal
+    await bot.change_presence(activity=discord.Game(name=f"m/help - {botver}")) # Change the bot activity
+
+@bot.command() # Help command. This will give you all of the commands
+async def help(ctx):
+    await ctx.send("""Minimal | By Cob:web Development 
+m/help                   - Shows this message
+m/about                  - Shows the bot statistics and ping
+m/calculate              - Calculate basic math. +-/*sqrt()sin()cos()
+m/warn <memberMention>   - Warn a member
+m/unwarn <memberMention> - Unwarn a member
+m/status <memberMention> - get warning status of member""")
+
+@bot.command() # About command. This includes; Bot latency, Bot guild number, Bot uptime, Bot version.
+async def about(ctx):
+    # Get the latency of the bot
+    latency = round(bot.latency * 100)  # round rounds a decimal number into the nearest whole number. bot.latency is given as a decimal. (bot.latency x 100) = the time in ms. 
+    
+    # Get the number of guilds the bot is in
+    guilds = len(bot.guilds) # Where len is length of the array for bot.guilds
+    
+    # Get the bot uptime
+    current_time = time.time() # Sets current time to the time.
+    difference = int(round(current_time - start_time)) # Takes away the current time from the start time (rounded)
+    botuptime = str(datetime.timedelta(seconds=difference)) # Calculates the bot uptime and displays the difference.
+    
+    # Send all the about statistics to the user
+    await ctx.send(f'About Minimal: \n\n Ping: {latency}ms \n Uptime: {botuptime} \n Version: {botver} \n\n Minimal is serving {guilds} servers. \n This bot was created by Adam Salt \n Made with discord.py `Created by Cob:web Development:` \n https://cob-web.xyz/discord/') # Shows all the output for the about command
+
+@bot.command() # calculate command
+async def calculate(ctx, *args):
+    command = ""
+    cont = False
+    for i in args:
+        for j in functions:
+            if j in i:
+                command += i
+                cont = True
+        if cont: 
+            cont = False
+            continue
+
+        try:
+            float(i)
+            command += i
+        except:
+            print("syntax error")
+            break
+
+    await ctx.send("the answer is: " + str(eval(command)))
+
+@bot.command(pass_context=True)
+async def warn(ctx, user: Member): # warning a member
+    role_names = map(lambda role : role.name, ctx.message.author.roles) # get user roles to a list
+    if "Moderator" in role_names:

```

Comment:
Rather than checking whether a user has a role exactly moderator, you should check for permissions of people's roles, such as Kick/Ban permissions or Administrator permissions